nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/QuerySObject.java

```diff
@@ -0,0 +1,491 @@
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor AGE_FIELD = new PropertyDescriptor.Builder()
+            .name("age-field")
+            .displayName("Age Field")
+            .description("The name of a TIMESTAMP field that will be used to limit all and filter already retrieved records."
+                    + " Only records that are older than the previous run time of this processor will be retrieved."
+            )
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor INITIAL_AGE_FILTER = new PropertyDescriptor.Builder()
+            .name("initial-age-filter")
+            .displayName("Initial Age Filter")
+            .description("When 'Age Field' is set the value of this property will serve as a filter when this processor runs the first time."
+                    + " Only records that are older than this value be retrieved."
+            )
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor AGE_DELAY = new PropertyDescriptor.Builder()
+            .name("age-delay")
+            .displayName("Age Delay")
+            .description("When 'Age Field' is set the age-based filter will be adjusted by this amount."
+                    + " Only records that are older than the previous run time of this processor, by at least this amount, will be retrieved."
+            )
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .defaultValue("5 s")
+            .build();
+
+    static final PropertyDescriptor BASE_URL = new PropertyDescriptor.Builder()
+            .name("salesforce-base-url")
+            .displayName("Base URL")
+            .description("")
+            .required(true)
+            .addValidator(Validator.VALID)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    static final PropertyDescriptor API_VERSION = new PropertyDescriptor.Builder()
+            .name("salesforce-api-version")
+            .displayName("API Version")
+            .description("")
+            .required(true)
+            .addValidator(Validator.VALID)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .defaultValue("46.0")
+            .build();
+
+    static final PropertyDescriptor AUTH_SERVICE = new PropertyDescriptor.Builder()
+            .name("auth-service")
+            .displayName("Auth service")
```

Comment:
This property name could be expanded for greater clarity. Adding a description would also be helpful.
```suggestion
            .displayName("OAuth2 Access Token Provider")
```